pallets/collator-assignment/src/lib.rs

```diff
@@ -177,13 +179,22 @@ pub mod pallet {
                 .into_iter()
                 .map(|(para_id, _)| para_id)
                 .collect();
+
+            // We read current assigned collators
+            let old_assigned = Self::read_assigned_collators();
+            let old_assigned_para_ids = old_assigned.container_chains.clone().into_keys().collect();
```

Comment:
This looks very similar but here you are only cloning the keys, while using `clone` clones both keys and values.

```suggestion
            let old_assigned_para_ids = old_assigned.container_chains.keys().cloned().collect();
```